### PR TITLE
[fwd port] check view consistency in fetch/exercise by interface 

### PR DESCRIPTION
**Pull request by paulbrauner-da**
Forward port of https://github.com/digital-asset/daml/pull/19857. Because `main` only has soft fetches, its code is quite different from `main-2.x`: there is no more `fetchAny` for instance. So the port is not a mere cherry-pick, I had to tweak it and it could benefit from a proper review.

**make sure to hide whitespace changes when reviewing**

### CODE REVIEW

File: sdk/daml-lf/interpreter/src/test/scala/com/digitalasset/daml/lf/speedy/SBuiltinInterfaceTest.scala [hunk 1]
```diff
@@ -24,28 +23,160 @@ import org.scalatest.Inside
...
+class SBuiltinInterfaceUpgradeTest extends AnyFreeSpec with Matchers with Inside {
+
+  import EvalHelpers._
+
+  val alice = Ref.Party.assertFromString("Alice")
 
-class SBuiltinInterfaceTest(majorLanguageVersion: LanguageMajorVersion)
+  // The following code defines a package -iface-pkg- that defines a single interface Iface.
+  val ifacePkgName = Ref.PackageName.assertFromString("-iface-pkg-")
+  val ifacePkgId = Ref.PackageId.assertFromString("-iface-package-id-")
```
Comment (remyhaemmerle-da) [inline sdk/daml-lf/interpreter/src/test/scala/com/digitalasset/daml/lf/speedy/SBuiltinInterfaceTest.scala:49]: extremely picky comment. 
why `-iface-pkg-` and `-iface-package-`

File: sdk/daml-lf/interpreter/src/test/scala/com/digitalasset/daml/lf/speedy/SBuiltinInterfaceTest.scala [hunk 2]
```diff
@@ -24,28 +23,160 @@ import org.scalatest.Inside
...
+
+  // All three of -iface-package-id-, -implem-pkg-id-1-, and -implem-pkg-id-2- are made available to the interpreter.
+  val compiledPackages = PureCompiledPackages.assertBuild(
+    Map(
+      ifacePkgId -> ifacePkg,
+      implemPkgId(1) -> implemPkg(1),
+      implemPkgId(2) -> implemPkg(2),
+    ),
+    // TODO: revert to the default compiler config once it supports upgrades
+    Compiler.Config.Dev(LanguageMajorVersion.V2),
```
Comment (remyhaemmerle-da) [inline sdk/daml-lf/interpreter/src/test/scala/com/digitalasset/daml/lf/speedy/SBuiltinInterfaceTest.scala:106]: can we factorize the version and the comment (lines 52, 82 and here) ?

**Review by remyhaemmerle-da — APPROVED** [review]
LGTM. Thanks.

File: sdk/daml-lf/interpreter/src/test/scala/com/digitalasset/daml/lf/speedy/SBuiltinInterfaceTest.scala (hunk 1 above)
Comment (paulbrauner-da) [reply sdk/daml-lf/interpreter/src/test/scala/com/digitalasset/daml/lf/speedy/SBuiltinInterfaceTest.scala:49]: Done

File: sdk/daml-lf/interpreter/src/test/scala/com/digitalasset/daml/lf/speedy/SBuiltinInterfaceTest.scala (hunk 2 above)
Comment (paulbrauner-da) [reply sdk/daml-lf/interpreter/src/test/scala/com/digitalasset/daml/lf/speedy/SBuiltinInterfaceTest.scala:106]: Done

**Pull request merged**